tiledb/api/src/array/domain.rs

```diff
@@ -55,26 +86,70 @@ impl<'ctx> Domain<'ctx> {
         ndim as usize
     }
 
-    pub fn dimension(&self, idx: usize) -> TileDBResult<Dimension<'ctx>> {
+    pub fn has_dimension<K: Into<DimensionKey>>(
+        &self,
+        key: K,
+    ) -> TileDBResult<bool> {
+        match key.into() {
+            DimensionKey::Index(idx) => Ok(idx < self.ndim()),
+            DimensionKey::Name(name) => {
```

Comment:
Oooh! Nice, this is a pattern to remember to avoid having to have multiple functions in a lot of the places where we allow lookups by index or name.